### PR TITLE
Update README for Azure Hosting projects to fix description of the me…

### PR DESCRIPTION
**Pull request by radical**
…thods to add the resources, and show AddConnectionString as an alternative to use an existing service
 ###### Microsoft Reviewers: [Open in CodeFlow](https://microsoft.github.io/open-pr/?codeflow=https://github.com/dotnet/aspire/pull/3732)

### CODE REVIEW

File: src/Components/Aspire.Azure.Data.Tables/README.md [hunk 1]
```diff
@@ -125,13 +125,15 @@ dotnet add package Aspire.Hosting.Azure.Storage
 Then, in the _Program.cs_ file of `AppHost`, add a Table Storage connection and consume the connection using the following methods:
 
 ```csharp
-var tables = builder.AddAzureStorage("storage").AddTables("tables");
+var tables = builder.ExecutionContext.IsPublishMode
+    ? builder.AddAzureStorage("storage").AddTables("tables")
+    : builder.AddAzureStorage("storage").AddConnectionString("tables");
```
Comment (eerhardt) [inline src/Components/Aspire.Azure.Data.Tables/README.md:130]:
```suggestion
    : builder.AddConnectionString("tables");
```

File: src/Components/Aspire.Azure.Messaging.EventHubs/README.md [hunk 2]
```diff
@@ -138,13 +138,15 @@ dotnet add package Aspire.Hosting.Azure.EventHubs
 Then, in the _Program.cs_ file of `AppHost`, add an Event Hubs connection and an Event Hub resource and consume the connection using the following methods:
 
 ```csharp
-var eventHubs = builder.AddAzureEventHubs("eventHubsConnectionName").AddEventHub("MyHub");;
+var eventHubs = builder.ExecutionContext.IsPublishMode
+    ? builder.AddAzureEventHubs("eventHubsConnectionName").AddEventHub("MyHub")
+    : builder.AddConnectionString("eventHubsConnectionName").AddEventHub("MyHub");
```
Comment (eerhardt) [inline src/Components/Aspire.Azure.Messaging.EventHubs/README.md:143]:
```suggestion
    : builder.AddConnectionString("eventHubsConnectionName");
```

File: src/Components/Aspire.Azure.Storage.Blobs/README.md [hunk 3]
```diff
@@ -125,13 +125,15 @@ dotnet add package Aspire.Hosting.Azure.Storage
 Then, in the _Program.cs_ file of `AppHost`, add a Blob Storage connection and consume the connection using the following methods:
 
 ```csharp
-var blobs = builder.AddAzureStorage("storage").AddBlobs("blobs");
+var blobs = builder.ExecutionContext.IsPublishMode
+    ? builder.AddAzureStorage("storage").AddBlobs("blobs")
+    ? builder.AddAzureStorage("storage").AddConnectionString("blobs");
```
Comment (eerhardt) [inline src/Components/Aspire.Azure.Storage.Blobs/README.md:130]:
```suggestion
    ? builder.AddConnectionString("blobs");
```

File: src/Components/Aspire.Azure.Storage.Queues/README.md [hunk 4]
```diff
@@ -125,13 +125,15 @@ dotnet add package Aspire.Hosting.Azure.Storage
 Then, in the _Program.cs_ file of `AppHost`, add a Storage Queue connection and consume the connection using the following methods:
 
 ```csharp
-var queue = builder.AddAzureStorage("storage").AddQueues("queue");
+var queue = builder.ExecutionContext.IsPublishMode
+    ? builder.AddAzureStorage("storage").AddQueues("queue")
+    ? builder.AddAzureStorage("storage").AddConnectionString("queue");
```
Comment (eerhardt) [inline src/Components/Aspire.Azure.Storage.Queues/README.md:130]:
```suggestion
    ? builder.AddConnectionString("queue");
```

**Review by eerhardt — COMMENTED** [review]
It might make sense to try each snippet locally in a playground app. Just to make sure it compiles.

**Review by mitchdenny — APPROVED** [review]

File: src/Components/Aspire.Azure.Storage.Blobs/README.md [hunk 5]
```diff
@@ -125,13 +125,15 @@ dotnet add package Aspire.Hosting.Azure.Storage
...
+var blobs = builder.ExecutionContext.IsPublishMode
+    ? builder.AddAzureStorage("storage").AddBlobs("blobs")
+    ? builder.AddConnectionString("blobs");
 
 var myService = builder.AddProject<Projects.MyService>()
                        .WithReference(blobs);
 ```
 
-The `AddBlobs` method will read connection information from the AppHost's configuration (for example, from "user secrets") under the `ConnectionStrings:blobs` config key. The `WithReference` method passes that connection information into a connection string named `blobs` in the `MyService` project. In the _Program.cs_ file of `MyService`, the connection can be consumed using:
+The `AddBlobs` method adds a Azure Storage blob resource to the builder. Or `AddConnectionString` method can be used be used to read connection information from the AppHost's configuration (for example, from "user secrets") under the `ConnectionStrings:blobs` config key. The `WithReference` method passes that connection information into a connection string named `blobs` in the `MyService` project. In the _Program.cs_ file of `MyService`, the connection can be consumed using:
```
Comment (eerhardt) [inline src/Components/Aspire.Azure.Storage.Blobs/README.md:136]:
```suggestion
The `AddBlobs` method adds an Azure Storage blob resource to the builder. Or `AddConnectionString` method can be used be used to read connection information from the AppHost's configuration (for example, from "user secrets") under the `ConnectionStrings:blobs` config key. The `WithReference` method passes that connection information into a connection string named `blobs` in the `MyService` project. In the _Program.cs_ file of `MyService`, the connection can be consumed using:
```

File: src/Components/Aspire.Azure.Storage.Queues/README.md [hunk 6]
```diff
@@ -125,13 +125,15 @@ dotnet add package Aspire.Hosting.Azure.Storage
...
+var queue = builder.ExecutionContext.IsPublishMode
+    ? builder.AddAzureStorage("storage").AddQueues("queue")
+    ? builder.AddConnectionString("queue");
 
 var myService = builder.AddProject<Projects.MyService>()
                        .WithReference(queue);
 ```
 
-The `AddQueues` method will read connection information from the AppHost's configuration (for example, from "user secrets") under the `ConnectionStrings:queue` config key. The `WithReference` method passes that connection information into a connection string named `queue` in the `MyService` project. In the _Program.cs_ file of `MyService`, the connection can be consumed using:
+The `AddQueues` method adds a Azure Storage queue to the builder. Or `AddConnectionString` can be used to read connection information from the AppHost's configuration (for example, from "user secrets") under the `ConnectionStrings:queue` config key. The `WithReference` method passes that connection information into a connection string named `queue` in the `MyService` project. In the _Program.cs_ file of `MyService`, the connection can be consumed using:
```
Comment (eerhardt) [inline src/Components/Aspire.Azure.Storage.Queues/README.md:136]:
```suggestion
The `AddQueues` method adds an Azure Storage queue to the builder. Or `AddConnectionString` can be used to read connection information from the AppHost's configuration (for example, from "user secrets") under the `ConnectionStrings:queue` config key. The `WithReference` method passes that connection information into a connection string named `queue` in the `MyService` project. In the _Program.cs_ file of `MyService`, the connection can be consumed using:
```

File: src/Components/Aspire.Azure.Data.Tables/README.md [hunk 7]
```diff
@@ -125,13 +125,15 @@ dotnet add package Aspire.Hosting.Azure.Storage
...
+var tables = builder.ExecutionContext.IsPublishMode
+    ? builder.AddAzureStorage("storage").AddTables("tables")
+    : builder.AddConnectionString("tables");
 
 var myService = builder.AddProject<Projects.MyService>()
                        .WithReference(tables);
 ```
 
-The `AddTables` method will read connection information from the AppHost's configuration (for example, from "user secrets") under the `ConnectionStrings:tables` config key. The `WithReference` method passes that connection information into a connection string named `tables` in the `MyService` project. In the _Program.cs_ file of `MyService`, the connection can be consumed using:
+The `AddTables` method will add a Azure Storage table resource to the builder. Or `AddConnectionString` can be used to read the connection information from the AppHost's configuration (for example, from "user secrets") under the `ConnectionStrings:tables` config key. The `WithReference` method passes that connection information into a connection string named `tables` in the `MyService` project. In the _Program.cs_ file of `MyService`, the connection can be consumed using:
```
Comment (eerhardt) [inline src/Components/Aspire.Azure.Data.Tables/README.md:136]:
```suggestion
The `AddTables` method will add an Azure Storage table resource to the builder. Or `AddConnectionString` can be used to read the connection information from the AppHost's configuration (for example, from "user secrets") under the `ConnectionStrings:tables` config key. The `WithReference` method passes that connection information into a connection string named `tables` in the `MyService` project. In the _Program.cs_ file of `MyService`, the connection can be consumed using:
```

**Review by eerhardt — APPROVED** [review]
Thanks

**Comment by danmoseley** [comment]
If you port to 8.0 I'll approve it

**Pull request merged**